src/parser.y

```diff
@@ -595,7 +595,7 @@ VariableDecl:
         Type DeclIdList ';' {
             CALL(@1, @3, type_pop());
         }
-        | Type error ';'
+        | Type error
```

Comment:
I am uneasy with this: if there is no `;`, then bison will not be able to recover from the error and continue, i.e. it will give up parsing and declare the rest of the input as error.

Reply:
From my testing this seems to give better results, I also added some tests to document this in test_parser.cpp

But consider the input:
```
int x;
int y // no semicolon
int z;
```
With the old grammar input would be ignored until the next semicolon which means int y and int z would be ignored.
However with the new grammar only int y is ignored, int z is picked up correctly

Also function definitions behave better as we can now recover on `}`